lib/commands/empty_bucket.go

```diff
@@ -0,0 +1,98 @@
+package commands
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/GetTerminus/s3helper/lib/parser"
+	"github.com/GetTerminus/s3helper/lib/provider"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/pkg/errors"
+)
+
+// EmptyBucketCommand represents the options that can be passed to the empty-bucket subcommand.
+type EmptyBucketCommand struct {
+	Bucket string `short:"b" long:"bucket" value-name:"bucket" description:"the bucket to empty" required:"true"`
+}
+
+func init() {
+	var cmd EmptyBucketCommand
+	parser.OptParser.AddCommand(
+		"empty-bucket",
+		"Erase all contents in an s3 bucket",
+		"Erase all objects and delete markers in an s3 bucket",
+		&cmd,
+	)
+}
+
+// Execute implements the interface for the go-flags subcommand.
+func (cmd *EmptyBucketCommand) Execute(args []string) error {
+	fmt.Fprintf(os.Stdout, "Deleting contents of s3://%s\n", cmd.Bucket)
+
+	// Create a session and get an s3 handle
+	session, err := provider.AWSSession()
+	if err != nil {
+		return errors.Wrap(err, "error getting aws session")
+	}
+
+	svc := s3.New(session)
+
+	totalDeleted := 0
+
+	for {
+		// 2 API calls are needed to delete things in s3, the first will get a list
+		// of the objects in the bucket, and the second to delete those things
+		resp, err := svc.ListObjectVersions(&s3.ListObjectVersionsInput{
```

Comment:
I think there is a V2 out there already

Reply:
at least i see ListObjectsV2Input, maybe not a v2 yet for everything

Reply:
ListObjectVersions does not have a V2 yet, using that method in case the bucket is a versioned bucket. That method will also list any delete markers in the bucket.